### PR TITLE
update assignment dto and linting and type fixed in doubt CRUD setup

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
1). linting  fixed and Object [] replaced by [] to match types in doubt CRUD set

2). updatedto for assignment added to for name to be the only(minimum) criteria for updation of assignment using a new data transfer object

As we can see that only name is required for updation

image

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 
pls suggest changes if any

File: src/modules/assignment/dto/update-assignment.dto.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { IsNotEmpty } from 'class-validator';
+
+export class UpdateAssignmentDTO {
+  @IsNotEmpty()
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/dto/update-assignment.dto.ts:4]: remove is empty

File: src/modules/assignment/dto/update-assignment.dto.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  readonly name: string;
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/dto/update-assignment.dto.ts:5]: Also add links in dto

According to schema defined: https://github.com/codeforcauseorg/edu-server/blob/HEAD/src/schemas/assignment.schema.ts

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/modules/assignment/dto/update-assignment.dto.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/dto/update-assignment.dto.ts:2]: @Abhishek-kumar09 
1). removed @IsNotEmpty() decorator
2). added link object in updatedto (data transfer object)

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09
1). removed @isnotempty() decorator
2). added link object in updatedto (data transfer object)

**Review by Abhishek-kumar09 — APPROVED** [review]

**Pull request merged**